Remotion/Web/Core/UI/Controls/WebMenuItem.cs

```diff
@@ -433,9 +440,9 @@ public virtual void LoadResources (IResourceManager resourceManager, IGlobalizat
       if (!string.IsNullOrEmpty (key))
         Category = resourceManager.GetString (key);
 
-      key = ResourceManagerUtility.GetGlobalResourceKey (Text);
+      key = ResourceManagerUtility.GetGlobalResourceKey (Text.GetValue());
       if (!string.IsNullOrEmpty (key))
-        Text = resourceManager.GetString (key);
+        Text = resourceManager.GetWebString (key, Text.Type);
```

Comment:
Do you have any good ideas how we can put those lines into a utility method?

```      key = ResourceManagerUtility.GetGlobalResourceKey (Text.GetValue());
      if (!string.IsNullOrEmpty (key))
        Text = resourceManager.GetWebString (key, Text.Type);```
I can only think of using a by-ref for the value type, which might be okay, but feels kind of strange to me.

Reply:
We want to convert the following lines into a utility not just the if and the assignment?

```c#
key = ResourceManagerUtility.GetGlobalResourceKey (Text.GetValue());
if (!string.IsNullOrEmpty (key))
  Text = resourceManager.GetWebString (key, Text.Type);
```

Since we have the assignment on an arbitrary property it gets a bit trick. This are the solutions I came up with:

 - Use the backing field as `ref` target and move the code into a helper method
   `resourceManager.TryGetWebString(Text, ref _text);`
 - Use `out` and a `bool` return to get simplify the code around the if: 
   `if (resourceManager.TryGetWebString(Text, out var result) Text = result;`
 - Use a delegate to set the property. Useful if we need the property setter

Reply:
yeah, it's the complete pattern that's interesting. doing just the `if` isn't all that helpful. 

And that's what I was afraid of - there's not really a good solution, is there. Missed the bit about Property vs Field here, too, so....

The delegate's expensive, I think. We'd need to pass the property as an expression and then get the getter and setter delegates or do direct invokes. So, just accept it like this and maybe create an analyzer for this pattern?